packages/amplify_authenticator/lib/src/widgets/form_fields.dart

```diff
@@ -715,3 +717,158 @@ class _ConfirmSignInFormFieldState extends State<ConfirmSignInFormField> {
         obscureText: _obscureText);
   }
 }
+
+class VerifyUserFormFieldGroup extends StatefulWidget {
+  const VerifyUserFormFieldGroup({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  _VerifyUserFormFieldGroupState createState() =>
+      _VerifyUserFormFieldGroupState();
+}
+
+class _VerifyUserFormFieldGroupState extends State<VerifyUserFormFieldGroup> {
+  String? _value;
+  late VerifyUserViewModel _verifyUserViewModel;
+  late ConfirmVerifyUserViewModel _confirmVerifyUserViewModel;
+  late List<String> unverifiedAttributeKeys;
+
+  // set value for the verify user & confirm verify user view model
+  void setUserAttributeKey(String? value) {
+    setState(() {
+      _value = value;
+      _verifyUserViewModel.setUserAttributeKey(value);
+      _confirmVerifyUserViewModel.setUserAttributeKey(value);
```

Comment:
*Note:* I am not a huge fan of this pattern of setting state on two different view models at once. This is how username is shared between sign up and confirm sign up. It might be worth exploring another way to do share state.

Reply:
Yeah, I think just making the keys part of the bloc state solves all this.